packages/server/src/course/course.controller.ts

```diff
@@ -525,64 +523,19 @@ export class CourseController {
     await this.courseService.removeUserFromCourse(userCourse);
   }
 
-  /* TODO: This is the old login registration form.
-  @Post('submit_course')
-  async submitCourse(@Body() body: SubmitCourseParams): Promise<void> {
-    if (body.password !== process.env.APPLY_PASSWORD) {
-      throw new UnauthorizedException(
-        ERROR_MESSAGES.courseController.invalidApplyURL,
-      );
-    }
-
-    const season = body.semester.split(' ')[0];
-    const year = parseInt(body.semester.split(' ')[1]);
-
-    const semester = await SemesterModel.findOne({
-      where: { season, year },
-    });
-    if (!semester)
-      throw new BadRequestException(
-        ERROR_MESSAGES.courseController.noSemesterFound,
-      );
-
-    let course = null;
-    try {
-      // create the submitted course
-      course = await CourseModel.create({
-        name: body.name,
-        coordinator_email: body.coordinator_email,
-        icalURL: body.icalURL,
-        semesterId: semester.id,
-        enabled: false,
-        pending: true,
-        timezone: body.timezone,
-      }).save();
-    } catch (err) {
-      console.error(err);
-      throw new HttpException(
-        ERROR_MESSAGES.courseController.createCourse,
-        HttpStatus.INTERNAL_SERVER_ERROR,
-      );
-    }
-
+  @Post('/register_courses')
+  @UseGuards(JwtAuthGuard)
+  @Roles(Role.PROFESSOR)
+  async registerCourses(
+    @Body() body: RegisterCourseParams[],
+    @UserId() userId: number,
+  ): Promise<void> {
     try {
-      // create CourseSectionMappings for each unique submitted section number
-      new Set(body.sections).forEach(async (section) => {
-        await CourseSectionMappingModel.create({
-          genericCourseName: body.name,
-          section,
-          courseId: course.id,
-        }).save();
-      });
+      await this.courseService.registerCourses(body, userId);
     } catch (err) {
```

Comment:
since you're just directly throwing this err you could just get rid of the whole try/catch